File: obligation-cordapp/java-source/src/test/java/net/corda/examples/obligation/flows/IssueObligationTests.java
package net.corda.examples.obligation.flows;

import net.corda.core.identity.Party;
import net.corda.core.transactions.SignedTransaction;
import net.corda.examples.obligation.Obligation;
import org.junit.Test;

import static net.corda.finance.Currencies.POUNDS;
import static net.corda.testing.internal.InternalTestUtilsKt.chooseIdentity;
import static org.junit.Assert.assertEquals;

public class IssueObligationTests extends ObligationTests {

    @Test
    public void issueNonAnonymousObligationSuccessfully() throws Exception {
        SignedTransaction stx = issueObligation(a, b, POUNDS(1000), false);

        network.waitQuiescent();

        Obligation aObligation = (Obligation) a.getServices().loadState(stx.getTx().outRef(0).getRef()).getData();
        Obligation bObligation = (Obligation) b.getServices().loadState(stx.getTx().outRef(0).getRef()).getData();

        assertEquals(aObligation, bObligation);
    }

    @Test
    public void issueAnonymousObligationSuccessfully() throws Exception {
        SignedTransaction stx = issueObligation(a, b, POUNDS(1000), true);

        Party aIdentity = chooseIdentity(a.getServices().getMyInfo());
        Party bIdentity = chooseIdentity(b.getServices().getMyInfo());

        network.waitQuiescent();

        Obligation aObligation = (Obligation) a.getServices().loadState(stx.getTx().outRef(0).getRef()).getData();
        Obligation bObligation = (Obligation) b.getServices().loadState(stx.getTx().outRef(0).getRef()).getData();

        assertEquals(aObligation, bObligation);

        Party maybePartyALookedUpByA = a.getServices().getIdentityService().requireWellKnownPartyFromAnonymous(aObligation.getBorrower());
        Party maybePartyALookedUpByB = b.getServices().getIdentityService().requireWellKnownPartyFromAnonymous(aObligation.getBorrower());

        assertEquals(aIdentity, maybePartyALookedUpByA);
        assertEquals(aIdentity, maybePartyALookedUpByB);

        Party maybePartyCLookedUpByA = a.getServices().getIdentityService().requireWellKnownPartyFromAnonymous(aObligation.getLender());
        Party maybePartyCLookedUpByB = b.getServices().getIdentityService().requireWellKnownPartyFromAnonymous(aObligation.getLender());

        assertEquals(bIdentity, maybePartyCLookedUpByA);
        assertEquals(bIdentity, maybePartyCLookedUpByB);
    }
}

File: obligation-cordapp/kotlin-source/src/test/kotlin/net/corda/examples/obligation/flows/IssueObligationTests.kt
package net.corda.examples.obligation.flows

import net.corda.examples.obligation.Obligation
import net.corda.finance.POUNDS
import net.corda.testing.internal.chooseIdentity
import org.junit.Test
import kotlin.test.assertEquals

class IssueObligationTests : ObligationTests() {

    @Test
    fun `Issue non-anonymous obligation successfully`() {
        val stx = issueObligation(a, b, 1000.POUNDS, anonymous = false)

        network.waitQuiescent()

        val aObligation = a.services.loadState(stx.tx.outRef<Obligation>(0).ref).data as Obligation
        val bObligation = b.services.loadState(stx.tx.outRef<Obligation>(0).ref).data as Obligation

        assertEquals(aObligation, bObligation)
    }


    @Test
    fun `issue anonymous obligation successfully`() {
        val stx = issueObligation(a, b, 1000.POUNDS)

        val aIdentity = a.services.myInfo.chooseIdentity()
        val bIdentity = b.services.myInfo.chooseIdentity()

        network.waitQuiescent()

        val aObligation = a.services.loadState(stx.tx.outRef<Obligation>(0).ref).data as Obligation
        val bObligation = b.services.loadState(stx.tx.outRef<Obligation>(0).ref).data as Obligation

        assertEquals(aObligation, bObligation)

        val maybePartyALookedUpByA = a.services.identityService.requireWellKnownPartyFromAnonymous(aObligation.borrower)
        val maybePartyALookedUpByB = b.services.identityService.requireWellKnownPartyFromAnonymous(aObligation.borrower)

        assertEquals(aIdentity, maybePartyALookedUpByA)
        assertEquals(aIdentity, maybePartyALookedUpByB)

        val maybePartyCLookedUpByA = a.services.identityService.requireWellKnownPartyFromAnonymous(aObligation.lender)
        val maybePartyCLookedUpByB = b.services.identityService.requireWellKnownPartyFromAnonymous(aObligation.lender)

        assertEquals(bIdentity, maybePartyCLookedUpByA)
        assertEquals(bIdentity, maybePartyCLookedUpByB)
    }
}
